### PR TITLE
Remove ambiguity in matrix norm description

### LINKED ISSUE #238: Matrix p-norm definition in docs is ambiguous
In the docs, the matrix norms defined in base are described simply as `p` norms.  Unfortunately, the nomeclature and the notation is ambiguous. It is unclear if Julia implements Holder p-norms (matrix norm induced by the vector p-norm) or Schatten p-norms (the vector p-norm of the singular values of the matrix).  Which p-norm one expects depends on their field of study (I myself was expecting Schatten p-norms).  It turns out Julia implements Holder p-norms.

I don't think the default needs to change, but clarifying the docs would be nice, at least where matrix norms are mentioned explicitly.

### LINKED ISSUE #12653: Matrix p-norm definition in docs is ambiguous
In the docs, the matrix norms defined in base are described simply as `p` norms.  Unfortunately, the nomeclature and the notation is ambiguous. It is unclear if Julia implements Holder p-norms (matrix norm induced by the vector p-norm) or Schatten p-norms (the vector p-norm of the singular values of the matrix).  Which p-norm one expects depends on their field of study (I myself was expecting Schatten p-norms).  It turns out Julia implements Holder p-norms.

I don't think the default needs to change, but clarifying the docs would be nice, at least where matrix norms are mentioned explicitly.

### PR DESCRIPTION
**Pull request by marcusps**
Fix JuliaLang/LinearAlgebra.jl#238

### CODE REVIEW

File: base/docs/helpdb.jl [hunk 1]
```diff
@@ -14645,8 +14645,7 @@ doc"""
            __precompile__(isprecompilable::Bool=true)
 
 Specify whether the file calling this function is precompilable.  If
-``isprecompilable`` is ``true``, then ``__precompile__`` throws an
-exception when the file is loaded by ``using``/``import``/``require``
+``isprecompilable`` is ``true``, then ``__precompile__`` throws an exception
```
Comment (stevengj) [inline base/docs/helpdb.jl]: I think you accidentally reverted the doc change in #12661

Comment (marcusps) [inline base/docs/helpdb.jl]: I have no idea how that could have happened (I used the web interface, and only edited the `norm` bit, but there it is). I'll fix it.

**Comment by stevengj** [comment]
I've always heard this just called the "induced norm", not the "Hölder norm".  Googling "induced holder norm" mainly turns up references to Hölder's inequality.  Can you give a couple references?

**Comment by marcusps** [comment]
I was going by Higham's ["Estimating the matrix p-norm", Numer. Math. 62, 539-555 (1992)](http://rd.springer.com/article/10.1007%2FBF01396242).  I suppose it is possible he was using that terminology to mean "the matrix norm induced by the Hölder/Lp norm on vectors".

I usually hear about induced/subordinate norms in a more general context, where any norm on a linear space can induce a norm on the set of linear transformations of that space.  The completely bounded norm (on linear transformations of matrices) is an example of a norm that is induced (by the Schatten norm on matrices), but it is not one of the "p-norms".

 Maybe the solution is something like

> For matrices, the matrix norm induced by the vector ``p``-norm is used, 
> where valid values of ``p`` are ``1``, ``2``, or ``Inf``. (Note that for sparse matrices, ``p=2`` is currently 
> not implemented.) Use :func:`vecnorm` to compute the Frobenius norm.

How does that sound?

**Comment by marcusps** [comment]
["Lectures on Finite Precision Computations" by Françoise Chaitin-Chatelin, Valérie Frayssé](https://books.google.com/books?id=kopEVFl6-kMC&printsec=frontcover&source=gbs_ge_summary_r&cad=0#v=onepage&q&f=false) uses the term "Hölder matrix norm" the same way Higham does, but this usage does seem to be somewhat unusual.  In  many places I see the Hölder/Lp vector norm on matrices being called the Hölder norm.

I do think the alternative I suggested above works best.

**Comment by stevengj** [comment]
Yes, that solution sounds fine.  I agree that the "induced operator norm" refers to the norm induced by *any* vector norm; in this context I meant the norm induced by the vector Lp norm.

**Comment by marcusps** [comment]
Thanks, @stevengj. The discussed change has been pushed.

**Pull request merged**